android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterInstrumentation.java

```diff
@@ -87,24 +88,30 @@ public void onCreate(Bundle args) {
       }
     }
 
-    new InteropTester(testCase, host, port, serverHostOverride, useTls, testCa,
-        androidSocketFactoryTls,
-        new InteropTester.TestListener() {
-          @Override
-          public void onPreTest() {
-          }
+    try {
+      new InteropTester(testCase, host, port, serverHostOverride, useTls, testCa,
+          androidSocketFactoryTls,
+          new InteropTester.TestListener() {
+            @Override
+            public void onPreTest() {
+            }
 
-          @Override
-          public void onPostTest(String result) {
-            Bundle bundle = new Bundle();
-            bundle.putString("grpc test result", result);
-            if (InteropTester.SUCCESS_MESSAGE.equals(result)) {
-              finish(0, bundle);
-            } else {
-              finish(1, bundle);
+            @Override
+            public void onPostTest(String result) {
+              Bundle bundle = new Bundle();
+              bundle.putString("grpc test result", result);
+              if (InteropTester.SUCCESS_MESSAGE.equals(result)) {
+                finish(0, bundle);
+              } else {
+                finish(1, bundle);
+              }
             }
           }
-        }
-    ).execute();
+      ).execute();
+    } catch (Throwable t) {
+      Bundle bundle = new Bundle();
+      bundle.putString("Exception encountered", t.toString());
```

Comment:
Might consider using Throwables.getStackTraceAsString, which preserves the causal chain and stack trace.